builddockerimageCUDA.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+############################################
+## Shell script to build Docker IBM aihwkit 
+## CUDA enabled docker image for your GPU
+############################################
+
+# Get CUDA_ARCH for your GPU from `nvidia-smi`
+CUDA_ARCH=$(nvidia-smi --query-gpu=compute_cap --format=csv | sed -n '2 p' | tr -d '.')
+
+# Get number of cpu threads for build
+NUM_CPU_THREADS=$(nproc)
+
+echo "Detected CUDA_ARCH is $CUDA_ARCH"
+echo "Detected NUM_CPU_THREADS is $NUM_CPU_THREADS"
+
+# Build Container
+docker build \
+--tag aihwkit:$CUDA_ARCH-cuda$CUDA_VER-ubuntu$UBUNTU_VER \
+--build-arg CUDA_ARCH=$CUDA_ARCH \
+--build-arg CUDA_VER=11.7 \
+--build-arg UBUNTU_VER=22.04 \
+--build-arg NUM_CPU_THREADS=$NUM_CPU_THREADS \
+--file CUDA.Dockerfile .
+
+echo "Docker image aihwkit:$CUDA_ARCH-cuda$CUDA_VER-ubuntu$UBUNTU_VER is built successfully"
```

Comment:
Can you add a reduced version of the script as documentation? Since the bulk of the script is the `docker build` command (and the temporary variables can also be evaluated inline), we can avoid the need for a separate file and at the same time provide users with some insight on how to tune the generation of the image if needed.

@maljoras might have better insights on its placement - my recommendation would be as a new block after [in-place-installation](https://github.com/IBM/aihwkit/blob/master/docs/source/advanced_install.rst#in-place-installation), ie.

```
...
Here, you need to make sure that the ``PYTHONPATH`` is set to the ``src``
sub-directory of the ahwkit base directory, e.g. by (when being in the base directory)::

    $ export PYTHONPATH=`pwd`/src:$PYTHONPATH


CUDA-enabled docker image
~~~~~~~~~~~~~~~~~~~~~~~~~

(some explanation on how to build the image, and a copy-pasteable `docker build` command)
...
```

Reply:
I think I have added all the necessary changes.